File: src/main/java/ru/mm/surv/codecs/webm/incubator/streamm/Stream.java
package ru.mm.surv.codecs.webm.incubator.streamm;

import ru.mm.surv.codecs.webm.event.EventSourceImpl;

public class Stream extends EventSourceImpl {

    private MovieFragment fragment;
    private int fragmentAge;

    private byte[] header;

    private String mimeType = "application/octet-stream";

    private boolean runs = true;

    public synchronized boolean isRunning() {
        return runs;
    }

    public synchronized void stop() {
        runs = false;
    }

    public synchronized int getFragmentAge() {
        return fragmentAge;
    }

    public synchronized MovieFragment getFragment() {
        return fragment;
    }

    public synchronized byte[] getHeader() {
        return header;
    }

    public synchronized void setHeader(byte[] newHeader) {
        header = newHeader;
    }

    public String getMimeType() {
        return mimeType;
    }

    public void setMimeType(String mimeType) {
        this.mimeType = mimeType;
    }

    public synchronized void pushFragment(MovieFragment newFragment) {
        if (fragmentAge == 0)
            postEvent(new ServerEvent(ServerEvent.INPUT_FIRST_FRAGMENT));
        fragment = newFragment;
        fragmentAge++;
    }
}

File: src/main/java/ru/mm/surv/codecs/webm/incubator/streamm/Stream.kt
package ru.mm.surv.codecs.webm.incubator.streamm

import ru.mm.surv.codecs.webm.event.EventSourceImpl

open class Stream : EventSourceImpl() {
    @get:Synchronized
    var fragment: MovieFragment? = null
        private set

    @get:Synchronized
    var fragmentAge = 0
        private set

    @get:Synchronized
    @set:Synchronized
    var header: ByteArray? = null
    var mimeType = "application/octet-stream"

    @get:Synchronized
    var isRunning = true
        private set

    @Synchronized
    fun stop() {
        isRunning = false
    }

    @Synchronized
    fun pushFragment(newFragment: MovieFragment?) {
        if (fragmentAge == 0) postEvent(ServerEvent(ServerEvent.INPUT_FIRST_FRAGMENT))
        fragment = newFragment
        fragmentAge++
    }
}